src/plugins/rule/schema.js

```diff
@@ -0,0 +1,143 @@
+// Copyright 2019 OpenStax Poland
+// Licensed under the MIT license. See LICENSE file in the project root for
+// full license text.
+
+import { Block, Text } from 'slate'
+
+const TYPES = ['rule', 'theorem', 'lemma', 'corollary', 'law', 'proposition']
+
+function normalizeRule(editor, error) {
+    const { code, child, key, node, index } = error
+
+    switch (code) {
+    // A child of different type was expected.
+    case 'child_type_invalid': {
+        // A text child could only have been added to an exercise by Slate in
+        // event that all other children were removed. In such case we just want
+        // to drop the exercise.
+        if (Text.isText(child)) {
+            editor.removeNodeByKey(node.key)
+            return
+        }
```

Comment:
It can also be added by importing wrong cnxml. Text should be either moved outside of parent or wrapped with paragraph and then wrapped with statement / proof / example

Reply:
Such CNXML wouldn't pass validation (since text isn't allowed in `<rule>`), so I don't think it's a concern.

Reply:
When someone imports cnxml when editing draft it is saved correctly on server even if there were errors when validating

Reply:
Such cases should be handled by the CNXML loader, because as this case shows it's impossible to have the same set of normalization rules handle both editing errors and import errors. And while it is currently impossible/not viable to normalize imports differently (it will be after migration to 0.5X, as `slate-html-serializer` is deprecated and we'll have to replace it with a custom solution, which I wanted to do for a long time), I still believe that we should focus on editing behaviour.